specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2025-03-03/gallery.json

```diff
@@ -2966,6 +3607,248 @@
+          "readOnly": true,
+          "$ref": "#/definitions/GalleryProvisioningState"
+        }
+      },
+      "required": [
+        "supportedOSType"
+      ],
+      "description": "Describes the properties of a gallery Script Definition."
+    },
+    "GalleryScriptVersion": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryScriptVersionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common-types/v1/common.json#/definitions/Resource"
+        }
+      ],
+      "description": "Specifies information about the gallery Script Version that you want to create or update."
+    },
+    "GalleryScriptVersionUpdate": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryScriptVersionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/UpdateResourceDefinition"
+        }
+      ],
+      "description": "Specifies information about the gallery Script Version that you want to update."
+    },
+    "GalleryScriptVersionProperties": {
+      "type": "object",
+      "properties": {
+        "publishingProfile": {
+          "$ref": "#/definitions/GalleryScriptVersionPublishingProfile"
+        },
+        "safetyProfile": {
+          "$ref": "#/definitions/GalleryScriptVersionSafetyProfile"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "#/definitions/GalleryProvisioningState"
+        },
+        "replicationStatus": {
+          "readOnly": true,
+          "$ref": "#/definitions/ReplicationStatus"
+        }
+      },
+      "required": [
+        "publishingProfile"
+      ],
+      "description": "Describes the properties of a gallery script version."
+    },
+    "GalleryScriptVersionPublishingProfile": {
+      "type": "object",
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/ScriptSource"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryArtifactPublishingProfileBase"
+        }
+      ],
+      "required": [
+        "source"
+      ],
+      "description": "The publishing profile of a gallery image version."
+    },
+    "GalleryScriptVersionSafetyProfile": {
+      "type": "object",
+      "properties": {},
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryArtifactSafetyProfileBase"
+        }
+      ],
+      "description": "The safety profile of the Gallery Script Version."
+    },
+    "ScriptSource": {
+      "type": "object",
+      "properties": {
+        "scriptLink": {
+          "type": "string",
+          "description": "Required. The link of the source script, it must be a readable storage blob with SAS URI or publicly accessible URI or managed identity enabled."
+        },
+        "parameters": {
```

Comment:
is there a max limit on the number of items ?

Reply:
no. This can be though of as Powershell script parameters or shell script parameters. There are no limits on number of parameters themselves, but limited by system resources like maximum length of command, memory etc.